### PR TITLE
[#5828] improvement(CLI): Slightly misleading error message when user is not specified in Gravitino CLI

### PR DESCRIPTION
**Pull request by Abyss-lord**
### What changes were proposed in this pull request?

Fix missing leading error message when user is not specified in Gravitino CLI, include all commands.

### Why are the changes needed?

Fix: #5828 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

```bash
bin/gcli.sh  user create --metalake demo_metalake
# Missing --user option.

bin/gcli.sh  user details --metalake demo_metalake
# Missing --user option.

bin/gcli.sh  user delete --metalake demo_metalake
# Missing --user option.

bin/gcli.sh  user list --metalake demo_metalake
# anonymous
```

### CODE REVIEW

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java [hunk 1]
```diff
@@ -476,6 +476,16 @@ protected void handleUserCommand() {
...
 
+    if (CommandActions.LIST.equals(command)) {
+      newListUsers(url, ignore, metalake).handle();
+      return;
+    }
+
+    if (Objects.isNull(user)) {
+      System.err.println("User is undefined. Please use --user to specify a user.");
+      return;
+    }
```
Comment (tengqm) [inline clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:487]: I would do this in a different way:
```java
  // A user is mandatory for all actions except for `list`.
  if (Object.isNull(user) && !CommandActions.LIST.equals(command) {
    System.err.println("....");
    return;
  }
```

The handler for `list` can be left as is.

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java [hunk 2]
```diff
@@ -476,6 +476,16 @@ protected void handleUserCommand() {
 
     Command.setAuthenticationMode(auth, userName);
 
+    if (CommandActions.LIST.equals(command)) {
+      newListUsers(url, ignore, metalake).handle();
+      return;
+    }
+
+    if (Objects.isNull(user)) {
```
Comment (justinmclean) [inline clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:484]: Why not use user == null?

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java [hunk 3]
```diff
@@ -476,6 +476,16 @@ protected void handleUserCommand() {
...
+      System.err.println("User is undefined. Please use --user to specify a user.");
```
Comment (justinmclean) [inline clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:485]: I'd keep in line with other error messages and use "Missing --user option." even if we were to use more verbose error messages I'd avoid saying "undefined".

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java (hunk 2 above)
Comment (Abyss-lord) [reply clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java]: fix, thank you Justin, plz review it again.

**Review by justinmclean — APPROVED** [review]
LGTM thanks!

**Pull request merged**